### DISCUSSION TITLE
How to erase part of a polygon

### DISCUSSION THREAD

**Discussion by Lynne42** [fabricjs/fabric.js#7206] [Q&A]
How to erase a part of the polygon, I used globalCompositeOperation:'xor', but it erases the background image set together, how to only the objects with overlapping scopes (not including the background image)

I want to display the content corresponding to the background image in the overlap

![image](https://user-images.githubusercontent.com/46595240/124938942-4e858e80-e03b-11eb-915b-7779cb4aea94.png)

https://codesandbox.io/s/canvas-spb3n

**Comment by asturur** [comment]
There are 2 ways of doing this.
One is with 2d effects, the other is understanding what shape would the new polygon have and modify the points.
Global composite operation is isolated to the group if the polygons are grouped. Did you try to group them together?

**Reply by Lynne42** [reply]
Thanks for the reply, I just came into contact with canvas and haven’t found a good way to use it.
After testing for a long time, I used the method of generating new polygons through union and intersection based on the shape of the polygon. This makes it easier to merge polygons and split polygons, which is convenient for generating image binarized matrix data for later use.
Using the 2d globalCompositeOperation effect, if our polygon has transparency, this transparency will be applied to the background image. I have not found a way to cancel it.

**Reply by Lynne42** [reply]
https://codepen.io/qlniu/pen/gOWLaOe

I found that in the end, we still need to use the globalCompositeOperation effect, for example, when dealing with digging holes (two graphics containing relationships)

I hope that the included graphics will not be displayed, and the background image will be displayed after the transparent processing, which is the same as the effect of example 1, but it will also process the background image and display the color of the body

Example 1 can be achieved. I set the background image to canvas through css, but the background cannot achieve the zoom effect.

I don’t know how to deal with it. I hope that the background image can zoom with the canvas, and when the graphic is xor, the overlapping part of the background image corresponds to the area displayed.

**Reply by asturur** [reply]
If you need to dig holes, you can still dump polygons and move to paths.
A path can have holes using one of the 2 winding rules.
